queue_block.py

```diff
@@ -109,87 +92,78 @@ def pop(self, grp="null", count=1, reload=False):
-            # check for uniqueness if property is set
-            try:
-                unique_val = self.uniqueness(signal)
-                self._logger.debug(
-                    "Testing uniqueness for signal: {}".format(unique_val)
-                )
-            except Exception as e:
-                unique_val = None
-                
-            if unique_val is not None:
-                for idx, sig in enumerate(self._queues[grp]):
-                    try:
-                        sig_val = self.uniqueness(sig)
-                    except Exception as e:
-                        sig_val = None
-                    if sig_val == unique_val:
-                        self._logger.debug(
-                            "Signal {} already in {}_queue".format(sig_val, grp)
-                        )
-                        if self.update:
-                            self._queues[grp][idx] = signal
-                        return
-
-            # pop one off the top of that queue if it's at capacity
-            if len(queue) == self.capacity:
-                self._logger.debug(
-                    "Pushing signal and capactity of {}_signal is full: {}".format(
-                        grp, self.capacity
+        # check for uniqueness if property is set
+        try:
+            unique_val = self.uniqueness(signal)
+            self._logger.debug(
+                "Testing uniqueness for signal: {}".format(unique_val)
+            )
+        except Exception as e:
+            unique_val = None
+
+        if unique_val is not None:
+            for idx, sig in enumerate(self._queues[grp]):
+                try:
+                    sig_val = self.uniqueness(sig)
+                except Exception as e:
+                    sig_val = None
+                if sig_val == unique_val:
+                    self._logger.debug(
+                        "Signal {} already in {}_queue".format(sig_val, grp)
                     )
+                    if self.update:
+                        self._queues[grp][idx] = signal
+                    return
+
+        # pop one off the top of that queue if it's at capacity
+        if len(queue) == self.capacity:
+            self._logger.debug(
+                "Pushing signal and capactity of {}_signal is full: {}".format(
+                    grp, self.capacity
                 )
-                # self.pop(grp) but without needing a new lock
-                if self.peek(grp):
-                    self._queues[grp] = self._queues[grp][1:]
+            )
+            self._queues[grp] = self._queues[grp][1:]
+
+        self._logger.debug("Appending signal to {}_queue".format(grp))
+        self._queues[grp].append(signal)
 
-            self._logger.debug("Appending signal to {}_queue".format(grp))
-            self._queues[grp].append(signal)
+    def _push_group(self, signals, group):
+        # lock the queue before appending
+        lock = self._get_lock(group)
+        with lock:
```

Comment:
Same as before.

Also, in `_get_lock`, can we switch to the `with` syntax? Doing the acquire and release on our own always gives me anxiety